nautobot/core/management/commands/generate_performance_test_endpoints.py

```diff
@@ -0,0 +1,269 @@
+        # One of the nautobot apps: nautobot.circuits, nautobot.dcim, and etc. if not is_app
+        # One of the plugins: example_app, and etc. if is_app
+        app_name = lookup_str_list[0] if is_app else lookup_str_list[1]
+
+        model = pattern.default_args.get("model", None)
+        if model:
+            app_name = model._meta.app_label
+
+        # No need to test the login and logout endpoints for performance testing
+        if app_name == "users":
+            if pattern.name in ["login", "logout"]:
+                url_pattern = f"/{pattern.pattern}"  # /login, /logout
+                view_name = f"{pattern.name}"  # login, logout
+                return url_pattern, view_name, is_api_endpoint
+
+        # Handle the special case where a view exist in the core app
+        # but its url pattern and view name does not include the prefix "/core" or "core:"
+        # ['nautobot', 'core', "views", "HomeView"]
+        # ['nautobot', 'core', "api", "views", "APIRootView"]
+        if app_name == "core":
+            if pattern.name in ["api-root", "api-status", "graphql-api"]:
+                is_api_endpoint = True
+                url_pattern = f"/api/{pattern.pattern}"  # /api/status
+                view_name = f"{pattern.name}"  # api-status
+                return url_pattern, view_name, is_api_endpoint
+            elif pattern.name in ["home", "about", "search", "worker-status", "graphql", "metrics"]:
+                url_pattern = f"/{pattern.pattern}"  # /home, /about, /search
+                view_name = f"{pattern.name}"  # home, about, search
+                return url_pattern, view_name, is_api_endpoint
+            elif pattern.name in [
+                "csv-import-fields",
+                "filtersetfield-list-lookupchoices",
+                "filtersetfield-retrieve-lookupvaluedomelement",
+            ]:
+                return None, None, is_api_endpoint
+
+        # Handle the special case first for Installed apps related view is nested under the extras app.
+        # ['nautobot', 'extras', 'plugins', 'views', 'InstalledAppsView']
+        if app_name == "extras" and "plugins" in lookup_str_list:
+            # We need special case handling to determine if the endpoint is an api endpoint as well for this view
+            view_class_name = lookup_str_list[-1]
+            if "API" in view_class_name:
+                is_api_endpoint = True
+            apps_or_plugins = "plugins" if "plugins" in pattern.name else "apps"
+
+            if is_api_endpoint:
+                url_pattern = f"/api/{apps_or_plugins}/{pattern.pattern}"  # /api/apps/installed-apps
+                view_name = f"{apps_or_plugins}-api:{pattern.name}"  # apps-api:apps-list
+            else:
+                url_pattern = f"/{apps_or_plugins}/{pattern.pattern}"  # /apps/installed-apps
+                view_name = f"{apps_or_plugins}:{pattern.name}"  # apps:apps_list
+
+            return url_pattern, view_name, is_api_endpoint
```

Comment:
If we had the rule enabled, pylint would probably complain about having so many different `return` points from one function. Can we replace various `if` with `elif` so as to just have one `return` at the end of the function?